spec/bottom up interraction.md

```diff
@@ -0,0 +1,119 @@
+
+The reason we wait for the the change to be committed is so that the transactions that add the signatures don’t get rejected by `check_tx` because they are referring to a non-existing checkpoint.
+
+The signing and sending of transactions happens in the [`broadcast`](https://github.com/consensus-shipyard/ipc/blob/specs/fendermint/vm/interpreter/src/fvm/broadcast.rs) module which fetches the current nonce of the validator, estimates the gas, performs retries, etc. Because it fetches the nonce for each submission, it cannot be used in parallel.
+
+<aside>
+💡 To submit transactions the validators of the subnet need to have an Ethereum account with sufficient tokens to cover the gas cost. They can use `fund` in order bring in tokens from the parent subnet. 
+
+The fact that validators have to pay to submit transactions to the subnet they validate may seem harsh; intuitively these should be free! But remember that a subnet can contain Byzantine validators who might abuse their privileges of free transaction submission. However, there is no reason why Fendermint couldn’t contain extra logic to compensate successful submissions for their costs.
+
+</aside>
+
+The signature transactions are sent to the child ledger, where they accumulate until a quorum is reached, that is, more that 2/3 of the total power of validators have signed the checkpoint.
+
+Note that by this time the CometBFT validators could be different, which is why signature submissions contain a Merkle proof that shows that the submitter was indeed part of the committee responsible for signing a past checkpoint.
+
+## Relayers
+
+Once a checkpoint has gathered a quorum of signatures in the child ledger, it can be picked up by a *relayer* and submitted in the form of a transaction to the parent subnet.
+
+Ideally we would like some redundancy in the number of validators, so there is no single point of failure.
+
+Relayers generally should be rewarded for their service. There are numerous opinions on how to implement rewards, each with their drawback:
+
+- *Only the first submitter gets a reward.* To achieve redundancy the reward would have to be multiple of the cost. Depending on how likely it is to beat the fastest relayer (a parent validator might insert their own transaction to steal the rewards), it might make it unprofitable for multiple validators to operate.
+- *The first N submitters get rewards.* It is easy for any relayer to submit N transactions in a Sybil attack to reap all rewards, hampering redundancy.
+- *All submitters in a fixed time period get equal share of a fixed reward.* This takes out the competition aspect, and discourages Sybil attacks because the reward doesn’t grow. It should lead to a dynamic equilibrium of the number of relayers. However if the fixed time window is too wide, it encourages freeloaders who just repeat the first submission, which would make it look like there is redundancy where there isn’t.
```

Comment:
I like this option because it implies that the goal should be that all honest validators share the burden of the cost of submitting a txn. ie, at equilibrium, `shared_reward / num_validators ~= txn_cost`.
 
 > it encourages freeloaders who just repeat the first submission
 
 Commit-reveal pattern could be helpful here?

Reply:
Yes, that's what I thought, at equilibrium the reward covers the cost. Any more joiners and somebody leaves; too many leave and it's profitable to join, so you can target a 3x redundancy by giving out 3x transaction cost. 

I haven't thought of a commit reveal pattern. It sounds promising; maybe the payload can be turned into a Merkle tree with a sender specific proof to show they have it, and then anyone can reveal the content a few blocks later?

Reply:
Another option: 
- Senders update a hash with the Payload and with a sender-chosen one-time-use "secret". 
- Sender posts a txn with this final hash (the "commitment"). 
- In the next time window, senders post the actual payload plus their one-time secret (the "reveal").
- The contract validates this second txn by trying to recreate the commitment (hash the sent payload w/ secret and comparing to content of first txn).

The downside is needing some intermediate state (one `bytes32` per sender) and multiple txns, but the first one is relatively inexpensive.

Reply:
I suppose you could make the first transaction also take the "reveal" from the prior time window. So, just one txn per window needed.